cli/azd/cmd/actions/action.go

```diff
@@ -17,4 +23,6 @@ func (a ActionFunc) Run(ctx context.Context) error {
 type Action interface {
 	// Run executes the CLI command.
 	Run(ctx context.Context) error
+	// Executes after Run. The action can  print the last message in the console as a summary after its execution.
+	PostRun(ctx context.Context, runResult error) error
```

Comment:
I don't feel like introducing a `PostRun` function is enough for our UX overhaul.  `Run` func should be sufficient and it should then force us into a single model for how we design our actions.

Reply:
`PostRun` is required for `composing commands`, like `azd up`.

When running `azd up`, we are running multiple command's actions (invoking `Run`).  And we don't want to print the `SUCCESS` message for every inner command. Instead, we want to allow `azd up` to define what's the ending message.

The `ending message` is the main reason of `PostRun`.

Reply:
Could we get away with something like `runInternal` (to be consumed by internal usage, not necessarily at the command layer) vs `Run`?

Reply:
I don't think we can @weikanglim .  The action is an interface, and we need an addition to the behavior of that interface without using `Run()`.

An alternative would be to pass an extra parameter to `Run()` which `Run()` would use to decide whether or not to call the `runInternal()` method.  That would be delegating the responsibility of showing the ending log to the command by making the command aware of how it is running (as part of a composite command or not).  But I don't think that's a good approach, as the command should not care about running modes.

Reply:
@vhvb1989  In composite actions, the action doesn't depend on the interface -- it depends on the concertion.

Reply:
@vhvb1989 If the spirit of the PR is to make incremental UX changes, and currently targeting only `init` progress, would you consider using a hacky workaround for `init` (like the `finalOutputRedirect`) in this PR?

I do think that `PostRun` is a step in the wrong direction, and hence why I'm a little uncomfortable with having the change across all actions,. I am largely thinking that `action` returning the output contract is what we want long-term, and does solve the UX issues we've identified, but we should explore alternatives and come up with some thinking here.

Reply:
Updated `PostRun` and changed it for making `Run` to return the ActionResult that is used to print a message